components/centraldashboard/app/api.ts

```diff
@@ -3,72 +3,100 @@ import express from 'express';
-  };
-}
+const IAP_HEADER = 'X-Goog-Authenticated-User-Email';
+const IAP_PREFIX = 'accounts.google.com:';
+
+export class Api {
+  private platformInfo: PlatformInfo;
 
-/** API routes exposing information from Kubernetes */
-export function api(
-    k8sService: KubernetesService,
-    metricsService?: MetricsService): express.Router {
-  return express.Router()
-      .get(
-          '/env-info',
-          async (_: express.Request, res: express.Response) => {
-            const [platform, user] = await Promise.all([
-              k8sService.getPlatformInfo(),
-              getUserInfo(),
-            ]);
-            const returnPayoad = {platform, user} as EnvironmentInfo;
-            res.json(returnPayoad);
-          })
-      .get(
-          '/metrics/:type((node|podcpu|podmem))',
-          async (req: express.Request, res: express.Response) => {
-            let interval = Interval.Last15m;
-            if (Interval[req.query.interval] !== undefined) {
-              interval = Number(Interval[req.query.interval]);
-            }
-            if (metricsService) {
-              switch (req.params.type) {
-                case 'node':
-                  res.json(
-                      await metricsService.getNodeCpuUtilization(interval));
-                  break;
-                case 'podcpu':
-                  res.json(await metricsService.getPodCpuUtilization(interval));
-                  break;
-                case 'podmem':
-                  res.json(await metricsService.getPodMemoryUsage(interval));
-                  break;
-                default:
+  constructor(
+      private k8sService: KubernetesService,
+      private metricsService?: MetricsService) {}
+
+  /** Retrieves and memoizes the PlatformInfo. */
+  private async getPlatformInfo(): Promise<PlatformInfo> {
+    if (!this.platformInfo) {
+      this.platformInfo = await this.k8sService.getPlatformInfo();
+    }
+    return this.platformInfo;
+  }
+
+  /**
+   * Retrieves user information from headers.
+   * Supports:
+   *  GCP IAP (https://cloud.google.com/iap/docs/identity-howto)
+   */
+  private getUser(req: express.Request): string {
+    let email = 'anonymous@kubeflow.org';
+    if (req.header(IAP_HEADER)) {
+      email = req.header(IAP_HEADER).slice(IAP_PREFIX.length);
+    }
+    return email;
+  }
+
+  /**
+   * Returns the Express router for the API routes.
+   */
+  routes(): express.Router {
+    return express.Router()
+        .get(
+            '/env-info',
+            async (req: express.Request, res: express.Response) => {
+              console.log('getting env-info');
+              const [platform, user, namespaces] = await Promise.all([
+                this.getPlatformInfo(),
+                this.getUser(req),
+                this.k8sService.getNamespaces(),
+              ]);
+              res.json({
+                platform,
+                user,
+                namespaces: namespaces.map((n) => n.metadata.name),
```

Comment:
Keep in mind this needs to be fresh as of 30 seconds, as per @kunmingg's suggestion. So we may need another API to keep this fresh, or use sockets

Reply:
What specifically needs to be fresh, the list of namespaces? In that case we still have the dedicated namespaces endpoint to get the latest content back.

Reply:
Yup, it was more an FYI, than an ask for change. I thought your commit message said we removed that endpoint.